src/mapblock.cpp

```diff
@@ -355,7 +355,7 @@ static void correctBlockNodeIds(const NameIdMapping *nimap, MapNode *nodes,
 	}
 }
 
-void MapBlock::serialize(std::ostream &os, u8 version, bool disk, int compression_level)
+void MapBlock::serialize(std::ostream &os, u8 version, bool disk, int compression_level, bool disk_inv)
```

Comment:
"What's the difference between disk and disk_inv?"  ~ the next developer who's working on this code.

How about introducing a new enum for optimizations, which then can be used among all serialization functions? Suggestion:

```
enum OptimizationOption {
	OO_NONE = 0,
	OO_INV_INCREMENTAL = 0x0001,
	OO_INV_META_SPARSE = 0x0002,
	OO_INV_META_HUMAN  = 0x0004, // Short, human-readable metadata fields
};
```

Reply:
The difference is that disk_inv may be true if disk is false when send_all_item_metadata is true or the client has map saving enabled.